library/src/main/java/com/alamkanak/weekview/WeekView.java

```diff
@@ -301,7 +301,7 @@ public boolean onSingleTapConfirmed(MotionEvent e) {
                 List<EventRect> reversedEventRects = mEventRects;
                 Collections.reverse(reversedEventRects);
                 for (EventRect eventRect : reversedEventRects) {
-                    if (eventRect.event.getId() != mNewEventId &&eventRect.rectF != null && e.getX() > eventRect.rectF.left && e.getX() < eventRect.rectF.right && e.getY() > eventRect.rectF.top && e.getY() < eventRect.rectF.bottom) {
+                    if (!eventRect.event.getId().equals(mNewEventId) && eventRect.rectF != null && e.getX() > eventRect.rectF.left && e.getX() < eventRect.rectF.right && e.getY() > eventRect.rectF.top && e.getY() < eventRect.rectF.bottom) {
```

Comment:
I would write mNewEventId.equals(eventRect.event.getId()) to avoid possible null-pointer exceptions on the id of an event (mNewEventId won't be null)